src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1325,6 +1253,166 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
             }
         }
 
+        private void RunAnalyzers(
+            TouchedFileLogger? touchedFilesLogger,
+            ref Compilation compilation,
+            ImmutableArray<DiagnosticAnalyzer> analyzers,
+            ImmutableArray<AdditionalText> additionalTextFiles,
+            ref CancellationTokenSource? analyzerCts,
+            ref bool reportAnalyzer,
+            ref AnalyzerDriver? analyzerDriver,
+            ref DiagnosticBag? analyzerExceptionDiagnostics,
+            DiagnosticBag diagnostics,
+            CompilerAnalyzerConfigOptionsProvider analyzerConfigProvider,
+            CancellationToken cancellationToken)
+        {
+            AnalyzerOptions analyzerOptions = CreateAnalyzerOptions(
+                   additionalTextFiles, analyzerConfigProvider);
+
+            if (!analyzers.IsEmpty)
+            {
+                analyzerCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                analyzerExceptionDiagnostics = new DiagnosticBag();
+
+                // PERF: Avoid executing analyzers that report only Hidden and/or Info diagnostics, which don't appear in the build output.
+                //  1. Always filter out 'Hidden' analyzer diagnostics in build.
+                //  2. Filter out 'Info' analyzer diagnostics if they are not required to be logged in errorlog.
+                var severityFilter = SeverityFilter.Hidden;
+                if (Arguments.ErrorLogPath == null)
+                    severityFilter |= SeverityFilter.Info;
+
+                // Determine if we should support artifact generators or not.  If we have an specified output path, then
+                // we will run artifact generators.  Otherwise, we won't bother as we have no place to put their files.
+                Action<string, Action<Stream>>? createArtifactStream = null;
+                if (!string.IsNullOrWhiteSpace(Arguments.GeneratedArtifactsOutputDirectory))
+                {
+                    createArtifactStream =
```

Comment:
Consider making this a local function for readability.

Reply:
done.